server/protocols/jmap/src/main/java/org/apache/james/jmap/model/Message.java

```diff
@@ -294,7 +285,8 @@ private static boolean hasAttachment(List<Attachment> attachments) {
                                Optional<String> textBody,
                                Optional<String> htmlBody,
                                ImmutableList<Attachment> attachments,
-                               ImmutableMap<BlobId, SubMessage> attachedMessages) {
+                               ImmutableMap<BlobId, SubMessage> attachedMessages,
+                               ImmutableMap<String, Boolean> keywords) {
```

Comment:
... Once again a comment I already did:

 - keywords should have type **Keywords**. You do a .asMap() in getKeywords method.
 - Remove **is\*** fields. In **isDraft** for example do a keywords.contains(DRAFT);

Avoid state duplication please...